lib/hammer_cli_katello/host_collection.rb

```diff
@@ -51,15 +51,46 @@ class InfoCommand < HammerCLIKatello::InfoCommand
     end
 
     class HostsCommand < HammerCLIKatello::ListCommand
-      resource :host_collections, :hosts
+      resource :hosts, :index
       command_name "hosts"
 
+      option "--id", "HOST_COLLECTION_ID", _("Host Collection ID")
+      option "--name", "HOST_COLLECTION_NAME", _("Host Collection Name")
+
+      validate_options do
+        host_collection_options = [:option_id, :option_name]
+        organization_options = [:option_organization_id, :option_organization_name, :option_organization_label]
+
+        any(*host_collection_options).required
+
+        if option(:option_name).exist?
+          any(*organization_options).required
+        end
+      end
+
+      def host_collection_options
+        {
+            "option_name" => option_name,
+            "option_organization_name" => option_organization_name,
+            "option_organization_id" => option_organization_id,
+            "option_organization_label" => option_organization_label
+        }
```

Comment:
Please let me know if there is a better way to do this.  This seems strange to recreate what I would think should already be there in some existing options object that I am unaware of.